promql/promqltest/test.go

```diff
@@ -264,6 +266,41 @@ func parseSeries(defLine string, line int) (labels.Labels, []parser.SequenceValu
 	return metric, vals, nil
 }
 
+func parseExpect(defLine string) (expectCmdType, expectCmd, error) {
+	expectParts := patExpect.FindStringSubmatch(strings.TrimSpace(defLine))
+	expCmd := expectCmd{}
+	if expectParts == nil {
+		return 0, expCmd, fmt.Errorf("invalid expect statement, must match `%s`", patExpect.String())
+	}
+	var (
+		mode            = expectParts[1]
+		hasOptionalPart = expectParts[2] != ""
+		matchAnyRegex   = regexp.MustCompile(`^.*$`)
```

Comment:
This will still compile on any function call. We need to move this variable initialization to the global section (around line 54) to compile it only once.

Reply:
Done.